src/docs/arch-checkpoint.dox

```diff
@@ -19,46 +21,71 @@ WT_CONNECTION::rollback_to_stable.
 
 # The checkpoint algorithm #
 
-A checkpoint can be broken up into 5 main stages:
+The checkpoint algorithm can be considered as a coordinator of multiple aspects of the system. It
+relies on the @ref arch-btree "btree subsystem" for traversing in memory pages, the @ref arch-block "block manager subsystem" for
+handling on-disk blocks and the @ref arch-metadata "metadata subsystem" for constructing and interpreting the metadata
+information.
 
-_The prepare stage:_
+A checkpoint can be broken up into the following stages:
+
+## Locking ##
+
+The first stage of checkpoint is to gather the necessary checkpoint locks. The most important being
+the \c checkpoint_lock. This lock ensures only one checkpoint takes place at time. More granular locks
```

Comment:
I suggest, instead of "More granular", just say "Other". The schema lock is also a "big hammer" lock so it isn't particularly more granular than the checkpoint lock. Since this is the arch guide, for this locking section, you might say that the checkpoint lock has the highest precedence of the locks. If multiple locks are needed, code needs to take the checkpoint lock first and then other locks.